Python/pymath.c

```diff
@@ -80,17 +80,13 @@ round(double x)
 }
 #endif /* HAVE_ROUND */
 
-static const unsigned int BitLengthTable[32] = {
-    0, 1, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4,
-    5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5
-};
+#include <limits.h>
+#ifdef _MSC_VER
+#  include <intrin.h>
+#  pragma intrinsic(_BitScanReverse)
+#  define __builtin_clz(VALUE) do { unsigned long clz = 0; _BitScanReverse (&clz, VALUE); clz; } while (0);
+#endif
 
 unsigned int _Py_bit_length(unsigned long d) {
-   unsigned int d_bits = 0;
-   while (d >= 32) {
-       d_bits += 6;
-       d >>= 6;
-   }
-   d_bits += BitLengthTable[d];
-   return d_bits;
+   return d ? CHAR_BIT * sizeof (d) - __builtin_clz (d) : 0;
```

Comment:
This is the wrong function. You want `__builtin_clzl`, not `__builtin_clz`. But in any case, this isn't going to work unless there's also a fallback for platforms that don't provide `__builtin_clz`.